include/real/real.hpp

```diff
@@ -638,12 +638,44 @@ namespace boost {
              *  @params: real_num: boost real number whose power is to be evaluated
              *  @params: exponent: power to which real_num is to be raised (needs to be an integer)
              *  @return: returns a new boost real whose value is real_num^exponent
-             *  @author: Kishan Shukla
+             *  @author: Kishan Shukla & Vikram Singh Chundawat
              */
 
-            static real power(real<T> real_num, real<T> exponent){
- 
-                return real(real_operation<T>(real_num._real_p, exponent._real_p, OPERATION::POWER));
+            static real power(real<T> real_num, real<T> power){
+                // checking whether the number is integer or not
+                static real<T> zero("0");
+                static real<T> one("1");
+                real<T> result;
+                auto itr = power.get_real_itr();
+                itr.iterate_n_times(itr.maximum_precision());
+
+                // power is of non integer type, then we will use logarithm and exponent to find power a^b = exp(b * log(a))
+                if (itr.get_interval().lower_bound != itr.get_interval().upper_bound ||
+                        (int) itr.get_interval().lower_bound.digits.size() > itr.get_interval().lower_bound.exponent) {
+                    /**
+                     * result = exp(exponent * log(real_num))
+                     * Warning: the result of non-integral power of a negative number is a complex number,
+                     * and we do not support complex numbers. So, that will generate error.
+                     * Note: The error will be generated by logarithm function.
+                     **/
+                    
+
+                    /**
+                     * Now, if number is negative, then logarithm function will check out and throw error
+                     **/
+                    result = real(real_operation<T>(real_num._real_p, zero._real_p, OPERATION::LOGARITHM));
+                    result = real(real_operation<T>(result._real_p, power._real_p, OPERATION::MULTIPLICATION));
+                    result = real(real_operation<T>(result._real_p, zero._real_p, OPERATION::EXPONENT));
+                    return result;
+                }
+                // if power is integer and negative
+                else if(itr.get_interval().lower_bound.positive == false){
```

Comment:
a number is negative if its both boundaries are negative, (checking if upper boundary is negative does the job). but the lower boundary as negative gives not mutch information about the number sign.

I think I'm missing something here, but if this is the case, an explanation comment in the code bould be good.

Reply:
That whole part is now removed. I think I should have checked for upper bound. No specific reason for checking lower bound :) .